test/flink_test.go

```diff
@@ -120,3 +142,139 @@ func (s *CLITestSuite) TestFlink_Autocomplete() {
 		s.runIntegrationTest(test)
 	}
 }
+
+func (s *CLITestSuite) TestFlinkShell() {
```

Comment:
Is the idea to replace all existing `flink shell` integration tests with this style of testing?

Reply:
I think it still makes sense to keep our existing more unit style tests since they are fast, easier to implement and enable us to test certain edge cases more easily. This is just to also have a test for actually starting up the binary and executing some commands without mocking everything.

Currently we're also missing a mock backend and just test local commands, I was planning to add a mock backend in the future to fake for example a `select` statement

Reply:
> Currently we're also missing a mock backend and just test local commands, I was planning to add a mock backend in the future to fake for example a select statement

This would be great. The end goal would be for our existing integration tests and the Flink integration tests to use the same framework... although the DevProd team has been trying to switch us over to contract tests lately.